lib/fig/protocol/artifactory.rb

```diff
@@ -0,0 +1,391 @@
+# coding: utf-8
+
+require 'net/http'
+require 'uri'
+
+require 'fig/logging'
+require 'fig/network_error'
+require 'fig/package_descriptor'
+require 'fig/protocol'
+require 'fig/protocol/netrc_enabled'
+
+require 'artifactory'
+
+module Fig; end
+module Fig::Protocol; end
+
+class Fig::ArtifactoryClient
+  attr_reader :upload_url, :download_url
+
+  @upload_url = ''
+  @download_url = ''
+  @creds = ''
+
+  def initialize(options = {})
+  end
+
+  def upload_directory(local_path)
+  end
+
+  def download_package()
+  end
+
+  def discover_remote_files(remote_path)
+    begin
+      # Use artifactory gem to browse the repository - search for files in the specific path
+      # Instead of wildcard search, search for files that match the remote path pattern
+      search_pattern = "#{remote_path}/*"
+      puts "  Searching for pattern: #{search_pattern}" if @verbose
+      puts "  In repository: #{@download_repo_key}" if @verbose
+      
+      top_host = @base_endpoint.split('/')
```

Comment:
`top_host` does not appear to be used, but also I can't seem to find where `@base_endpoint` is assigned

Reply:
That whole class just got removed.